crates/l2/proposer/prover_server.rs

```diff
@@ -393,64 +531,147 @@ impl ProverServer {
+        // function verify(uint256,bytes,bytes32,bytes32,bytes32,bytes,bytes)
+        // blockNumber, blockProof, imageId, journalDigest, programVKey, publicValues, proofBytes
         // From crates/l2/contracts/l1/interfaces/IOnChainProposer.sol
-        let verify_proof_selector = keccak(b"verify(uint256,bytes,bytes32,bytes32)")
+        let mut calldata = keccak(b"verify(uint256,bytes,bytes32,bytes32,bytes32,bytes,bytes)")
             .as_bytes()
             .get(..4)
             .ok_or(ProverServerError::Custom(
                 "Failed to get verify_proof_selector in send_proof()".to_owned(),
             ))?
             .to_vec();
-        calldata.extend(verify_proof_selector);
 
         // The calldata has to be structured in the following way:
         // block_number
-        // size in bytes
-        // image_id digest
-        // journal digest
-        // size of seal
-        // seal
+        // offset of first bytes parameter
+        // image_id
+        // journal
+        // programVKey
+        // offset of second bytes parameter
+        // offset of third bytes parameter
+        // size of block_proof
+        // block_proof
+        // size of publicValues
+        // publicValues
+        // size of proofBytes
+        // proofBytes
 
         // extend with block_number
         calldata.extend(H256::from_low_u64_be(block_number).as_bytes());
 
-        // extend with size in bytes
-        // 4 u256 goes after this field so: 0x80 == 128bytes == 32bytes * 4
-        calldata.extend(H256::from_low_u64_be(4 * 32).as_bytes());
+        // extend with offset in bytes
+        calldata.extend(H256::from_low_u64_be(7 * 32).as_bytes());
 
         // extend with image_id
-        calldata.extend(image_id.as_bytes());
+        calldata.extend(risc0_contract_data.image_id);
 
         // extend with journal_digest
-        calldata.extend(journal_digest.as_bytes());
-
-        // extend with size of seal
-        calldata.extend(
-            H256::from_low_u64_be(seal.len().try_into().map_err(|err| {
-                ProverServerError::Custom(format!("Seal length does not fit in u64: {}", err))
-            })?)
-            .as_bytes(),
-        );
-        // extend with seal
-        calldata.extend(seal);
-        // extend with zero padding
-        let leading_zeros = 32 - ((calldata.len() - 4) % 32);
-        calldata.extend(vec![0; leading_zeros]);
+        calldata.extend(risc0_contract_data.journal_digest);
+
+        // extend with program_vkey
+        calldata.extend(sp1_contract_data.vk);
+
+        // extend with offset in bytes of second bytes parameter
+        let block_proof_len: u64 =
+            risc0_contract_data
+                .block_proof
+                .len()
+                .try_into()
+                .map_err(|err| {
+                    ProverServerError::Custom(format!(
+                        "calldata length does not fit in u64: {}",
+                        err
```

Comment:
```suggestion
                        "calldata length does not fit in u64: {err}"
```

Reply:
Ditto for the rest of the function.

Reply:
Done in [c7aa3b7](https://github.com/lambdaclass/ethrex/pull/1466/commits/c7aa3b7938549869411b412f53caf64eac0bbb2e)